Added signup feature.

frandiox opened the pull request:
@masahirotanaka This is working for s.monaca.mobi. I will update the URL and remove `rejectUnauthorized: false` when the API for monaca.mobi is updated. I'm not changing the language of the request (always `/en/api`) since the CLI is not translated to Japanese. Also, I'd probably need to implement some logic or add dependencies to detect the language.

 Do you think it's fine like this?

frandiox committed:
Added signup feature.

masahirotanaka commented:
@frankdiox Thanks. I want the hostname to set to `monaca.mobi`. Also, the URL can be fixed to `/en/`, but I want to add `lang` parameter and `client` to the POST parameters.
- `lang`: Language
- `client`: Client type (specified by the client, CLI will set to `cli`)

frandiox commented:
@masahirotanaka Login function includes `language`, `clientType` (by default 'local'), `version` and `os`. Should I include those to be consistent or you prefer `lang` and `client` with 'cli' value?

frandiox commented:
@masahirotanaka Actually, @asialmakki added `from_cli: 1` parameter to the API. Perhaps we can use `clientType: 'local'` instead?

masahirotanaka commented:
Ah, I see. Then no need to add `client`.
I'll ask makki to see `clientType` and `version` instead.

frandiox commented:
@masahirotanaka So I added the new the same parameters that `login` includes and updated the API URL to the main server. It should work well when the main server is updated.

Something different: will we at some point send a code in the email that can be directly pasted into the CLI to activate the account? Or a link that doesn't need to input username and password again?

masahirotanaka commented:
Thanks @frankdiox. Saw the code.

The overall looks very good. Let us discuss the activation piece with @erisu tomorrow. I guess it is all right to just sending an email with the activation link, and when the user clicks on it, they are notified to return back to the CLI to proceed.
